zeebe/gateway-grpc/src/main/java/io/camunda/zeebe/gateway/interceptors/impl/AuthenticationHandler.java

```diff
@@ -87,15 +92,26 @@ public Either<Status, Context> authenticate(final String authorizationHeader) {
                 .augmentDescription("Failed to load OIDC principals, see cause for details")
                 .withCause(e));
       }
+      var context = Context.current();
+      if (oidcAuthenticationConfiguration.getGroupsClaim() != null) {
+        try {
+          context = context.withValue(GROUPS_CLAIMS, oidcGroupsLoader.load(token.getClaims()));
+        } catch (final Exception e) {
+          return Either.left(
+              Status.UNAUTHENTICATED
+                  .augmentDescription("Failed to load OIDC groups, see cause for details")
+                  .withCause(e));
+        }
+      }
```

Comment:
🙃 Bit of a nitpick, but maybe we can move this below the principal logic so that's not separated, or even better pull the principal and group stuff in a separate method.